### PR TITLE
feat: general dump-server headers as arg and improved cli building

### PR DESCRIPTION
**Pull request by melancholiai**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✔                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✔                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✖                                                                       |

### CODE REVIEW

File: src/commands/create/checks/index.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    const argument = 'state-bucket-name';
+    throw new CheckError(`${argument} is required when include-state is true`, argument, stateBucketName);
+  }
+  return true;
+};
+
+export const dumpServerUriCheck: CheckFunc<CreateArguments> = (argv) => {
+  const { dumpServerEndpoint } = argv;
+  if (dumpServerEndpoint !== undefined && isWebUri(dumpServerEndpoint) === undefined) {
+    const argument = 'dump-server-endpoint';
```
Comment (CptSchnitz) [inline src/commands/create/checks/index.ts]: as this is a constant extract to outer scope

File: src/commands/create/checks/index.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    throw new CheckError(`${argument} is not a valid web uri`, argument, dumpServerEndpoint);
+  }
+  return true;
+};
+
+export const headersCheck: CheckFunc<CreateArguments> = (argv) => {
```
Comment (CptSchnitz) [inline src/commands/create/checks/index.ts]: change to httpHeaders

File: src/commands/create/create.ts [hunk 3]
```diff
@@ -43,15 +61,16 @@ export class CreateCommand implements CommandModule<Argv, CreateArguments> {
         nargs: 1,
         type: 'string',
       })
-      .option('dumpServerToken', {
-        alias: ['tkn', 'dump-server-token'],
-        description: 'The token of the dump-server used for upload',
-        nargs: 1,
+      .option('dumpServerHeaders', {
+        alias: ['dsh', 'dump-server-headers'],
```
Comment (CptSchnitz) [inline src/commands/create/create.ts]: BIG H

File: src/commands/create/create.ts [hunk 4]
```diff
@@ -106,42 +103,44 @@ export class CreateCommand implements CommandModule<Argv, CreateArguments> {
...
-      const dumpMetadata = await this.manager.buildDumpMetadata(dumpMetadataOptions, s3BucketName);
+      const dumpMetadata = await manager.buildDumpMetadata(dumpMetadataOptions, s3BucketName);
 
       const pgDumpName = `${dumpMetadata.name}.${PG_DUMP_FILE_FORMAT}`;
-      const pgDumpPath = await this.manager.createPgDump(pgDumpName);
+      const pgDumpPath = await manager.createPgDump(pgDumpName);
 
       if (isS3Locked) {
-        await this.manager.unlockS3(stateBucketName);
+        await manager.unlockS3(stateBucketName as string);
```
Comment (CptSchnitz) [inline src/commands/create/create.ts:131]: add a comment explaining why the casting is correct

**Review by CptSchnitz — CHANGES_REQUESTED** [review]

File: src/commands/create/create.ts (hunk 4 above)
Comment (melancholiai) [reply src/commands/create/create.ts:131]: done.

File: src/commands/create/create.ts (hunk 3 above)
Comment (melancholiai) [reply src/commands/create/create.ts]: done.

File: src/commands/create/checks/index.ts (hunk 2 above)
Comment (melancholiai) [reply src/commands/create/checks/index.ts]: done.

File: src/commands/create/checks/index.ts (hunk 1 above)
Comment (melancholiai) [reply src/commands/create/checks/index.ts]: done.

**Review by CptSchnitz — APPROVED** [review]

**Pull request merged**